flake.nix

```
@@ -7,64 +7,95 @@
+              };
+            };
           };
         };
 
-        config = lib.mkIf cfg.enable {
-          assertions = [
-            {
-              assertion = config.security.sudo.enable;
-              message = "garnix.persistence needs security.sudo enabled, but some other module forced the value to 'false'";
-            }
-            {
-              assertion = config.security.sudo.execWheelOnly;
-              message = "garnix.persistence needs security.sudo.execWheelOnly enabled, but some other module forced the value to 'false'";
-            }
-            {
-              assertion = !config.security.sudo.wheelNeedsPassword;
-              message = "garnix.persistence needs security.sudo.wheelNeedsPassword disabled, but some other module forced the value to 'true'";
-            }
-            {
-              assertion = lib.elem "garnix" config.nix.settings.trusted-users;
-              message = "garnix.persistence needs the user 'garnix' to be present in 'nix.settings.trusted-users', but some other module forced it out.";
-            }
-            {
-              assertion = lib.elem "garnix" (lib.attrNames config.users.users);
-              message = "garnix.persistence needs the user 'garnix' to be present in 'users.users', but some other module forced it out.";
-            }
-          ];
-          services.openssh.enable = true;
-
-          security.sudo = {
-            enable = true;
-            execWheelOnly = true;
-            wheelNeedsPassword = false;
-          };
+        config = lib.mkMerge [
+          (lib.mkIf cfg.enable {
+            assertions = [
+              {
+                assertion = config.fileSystems."/".device == "/dev/sda1";
+                message = "garnix.server needs the fileSystems.\"/\".device to be \"/dev/sda1\"";
+              }
+              {
+                assertion = config.fileSystems."/".fsType == "ext4";
+                message = "garnix.server needs the fileSystems.\"/\".fsType to be \"ext4\"";
+              }
+              {
+                assertion = config.boot.loader.grub.derice == "/dev/sda";
+                message = "garnix.server needs the fileSystems.\"/\".fsType to be \"ext4\"";
+              }
+            ];
 
-          nix.settings.trusted-users = [ "garnix" ];
+            fileSystems."/" = {
+              device = "/dev/sda1";
+              fsType = "ext4";
+            };
+            boot.loader.grub.device = "/dev/sda";
+          })
 
-          users.users.garnix = {
-            description = "A user garnix uses for redeploying ${cfg.name}";
-            isNormalUser = true;
-            createHome = false;
-            openssh.authorizedKeys.keys = [
-              "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIJTPguYAuqp6qCU43u8g2hgWz4MLCEPPyoVPYO53qB+t garnixServer@garnix.io"
+          (lib.mkIf (cfg.enable && cfg.persistence.enable) {
```

Why does this depend on `cfg.enable`? If it needs `cfg.enable`, maybe there should be an assertion that that's switched on? Or it should just be independent, i.e:
```suggestion
          (lib.mkIf (cfg.persistence.enable) {
```

---

I can't really think of an example where you'd want `garnix.server.persistence.enable` and not `garnix.server.enable`, but yeah, an assertion would be better.

Do you think warnings would be better than errors, in this case?

What about in all other cases we use errors with assertions? Maybe we should allow users to override/go on at their own risk?

---

> Do you think warnings would be better than errors, in this case?
> 
> What about in all other cases we use errors with assertions? Maybe we should allow users to override/go on at their own risk?

Given that I sometimes ignore warnings in certain nix projects, because they are a lot, I think maybe errors are better? Not super opinionated though.